atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/ReadOnlyTransactionManager.java

```diff
@@ -161,6 +162,12 @@ public TransactionAndImmutableTsLock setupRunTaskWithConditionThrowOnConflict(Pr
         throw new UnsupportedOperationException("Not supported on this transaction manager");
     }
 
+    @Override
+    public List<TransactionAndImmutableTsLock> setupRunTaskWithConditionThrowOnConflictBatch(
+            PreCommitCondition condition, int count) {
```

Comment:
as in the other case, the PreCommitCondition is likely (although not guaranteed) to be identical between all requests. So I'd recommend you instead pass a _list_ of PreCommitConditions.

Reply:
You absolutely need to pass a list of them.